scripts/qa_crowbarsetup.sh

```diff
@@ -3069,22 +3089,34 @@ function onadmin_rebootcloud()
     pre_hook $FUNCNAME
     get_novacontroller
 
-    local cmachines=$(get_all_discovered_nodes)
+    local cmachines
     local m
+
+    if [[ $hacloud = 1 ]] ; then
+        cluster_node_assignment
+
+        for c in data network services; do
+            local clusternodes_var=$(echo clusternodes${clusternameservices})
+            for m in ${!clusternodes_var}; do
+                power_cycle_and_wait $m
+                m_hostname=$(echo $m | cut -d '.' -f 1)
+                wait_for 400 5 "crowbar node_state status | grep $m_hostname | grep -qiE \"ready$|problem$\"" "node $m_hostname to be online"
+            done
+            complain_if_problem
+        done
+
+        cmachines=$unclustered_nodes
```

Comment:
All you need to do is decouple this `cmachines` assignment from this `if` block.  And it seems clear to me that the code to calculate compute nodes should be independent of the code for rebooting clustered nodes.  Also a good opportunity to rename `cmachines` to something less cryptic, like `compute_nodes`.